firehose-source/src/main/java/source/TupleFactory.java

```diff
@@ -0,0 +1,174 @@
+        TupleBuilder httpTuple = TupleBuilder.tuple()
+                .put("startTime",httpStartStop.getStartTimestamp())
+                .put("stopTime",httpStartStop.getStopTimestamp())
+                .put("peerType",httpStartStop.getPeerType().toString())
+                .put("method",httpStartStop.getMethod().toString())
+                .put("uri",httpStartStop.getUri())
+                .put("remoteAddress",httpStartStop.getRemoteAddress())
+                .put("userAgent",httpStartStop.getUserAgent())
+                .put("statusCode", httpStartStop.getStatusCode())
+                .put("contentLength", httpStartStop.getContentLength())
+                .put("requestId", parseUUID(httpStartStop.getRequestId()));
+
+
+        String applicationId = parseUUID(httpStartStop.getApplicationId());
+        if(applicationId != null){
+            httpTuple.put("applicationId",applicationId);
+        }
+        String parentRequestId = parseUUID(httpStartStop.getParentRequestId());
+        if(parentRequestId != null){
+            httpTuple.put("parentRequestId", parentRequestId);
+        }
+        Tuple tuple = createBuilder(envelope)
+                .put("metrics",httpTuple.build())
+                .build();
+        return tuple;
+    }
+
+    private static Tuple fromValue(EventFactory.Envelope envelope){
+        MetricFactory.ValueMetric valueMetric = envelope.getValueMetric();
+        Tuple tuple = createBuilder(envelope).put(valueMetric.getName(), valueMetric.getValue()).put("unit",valueMetric.getUnit()).build();
+        return tuple;
+    }
+
+    private static TupleBuilder createBuilder(EventFactory.Envelope envelope){
+        return TupleBuilder.tuple()
+                .put("timestamp",envelope.getTimestamp())
+                .put("origin",envelope.getOrigin())
+                .put("host",envelope.getIp())
+                .put("index",envelope.getIndex())
+                .put("type",envelope.getEventType().toString())
+                ;
+    }
+
+    /**
+     * Needed due dropsonde decision to use uint64 and little endianess
+     * @param uuid
+     * @return
+     */
+    public static String parseUUID(UuidFactory.UUID uuid){
+
+        String uuidStr = null;
+
+        try{
+            BigInteger low = new BigInteger(String.valueOf(uuid.getLow()));
+            BigInteger high = new BigInteger(String.valueOf(uuid.getHigh()));
+            byte[] lowBytes = reverse(low.toByteArray());
+            byte[] highBytes = reverse(high.toByteArray());
+            byte[] data1 = new byte[lowBytes.length+highBytes.length];
+            System.arraycopy(lowBytes,0,data1,0,lowBytes.length);
+            System.arraycopy(highBytes, 0, data1, lowBytes.length, highBytes.length);
+
+            StringBuilder builder = new StringBuilder(Hex.encodeHexString(data1));
+            uuidStr = builder.insert(8,"-").insert(13,"-").insert(18,"-").insert(23,"-").toString();
+        }catch (Exception ex){
```

Comment:
The whole endianness is a nightmare. I looked all over to find a really cleaner way to do that, it does not help the fact that dropsonde is using uint64 and only in java 8 we had long support for that.

I'd happily flame this method if we have a better way. Please let me know and I'll change this. I got stuck for half a day on this madness when I created the old XD firehose.